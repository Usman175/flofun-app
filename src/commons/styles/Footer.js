import EStyleSheet from 'react-native-extended-stylesheet'

const styles = EStyleSheet.create({
  root: {
    flex: 0.1,
    backgroundColor: '$purpleColor',
    flexDirection: 'row',
  },
  bottomRight: {
    flex: 0.2,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  bottomLeft: {
    flex: 0.8,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: '2%',
  },
})

export default styles
