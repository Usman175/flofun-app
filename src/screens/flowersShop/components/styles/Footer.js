import EStyleSheet from 'react-native-extended-stylesheet'

const styles = EStyleSheet.create({
  bottomContainer: {
    flex: 0.1,
    backgroundColor: '$purpleColor',
    flexDirection: 'row',
  },
  bottomText: {
    color: '$whiteColor',
    fontSize: 16,
    fontFamily: 'montserratMedium',
  },
  bottomLeft: {
    flex: 0.8,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  bottomRight: {
    flex: 0.2,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  circle: {
    width: 80,
    height: 80,
    borderRadius: 40,
    borderColor: '$blackColor',
    borderWidth: 3,
    margin: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  selectedImage: {
    width: 76,
    height: 76,
    borderRadius: 38,
  },
  closeIcon: {
    position: 'absolute',
    top: 1,
    left: 6,
    backgroundColor: 'rgba(255, 255, 255, 0)',
  },
})

export default styles
