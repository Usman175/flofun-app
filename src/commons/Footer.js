import React, { Component } from 'react'
import { View, TouchableOpacity } from 'react-native'
import { FontAwesome } from '@expo/vector-icons'

import styles from './styles/Footer'

export default class Footer extends Component {
  render() {
    const { nextCallback } = this.props

    return (
      <View style={styles.root}>
        <View style={styles.bottomLeft} />
        <View style={styles.bottomRight}>
          <TouchableOpacity onPress={() => nextCallback()}>
            <FontAwesome
              name="arrow-right"
              size={40}
              color="#fff"
            />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
