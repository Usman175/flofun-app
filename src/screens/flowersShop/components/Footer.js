import React, { Component } from 'react'
import { View, TouchableOpacity, Image, TouchableWithoutFeedback, Animated } from 'react-native'
import { FontAwesome, MaterialCommunityIcons } from '@expo/vector-icons'

import styles from './styles/Footer'

export default class Footer extends Component {
  componentWillUpdate() {
    this.position = new Animated.ValueXY({ x: 0, y: 80 })

    Animated.spring(this.position, {
      toValue: { x: 0, y: 0 },
      duration: 500,
    }).start()
  }

  render() {
    const { selectedFlowers, removeSelectedFlower, navigate } = this.props
    return (
      <View style={styles.bottomContainer}>
        <View style={styles.bottomLeft}>
          {selectedFlowers.map((flower, key) => {
            const lastElement = key === selectedFlowers.length - 1

            if (lastElement) {
              return (
                <Animated.View key={flower._id} style={[this.position.getLayout(), styles.circle]}>
                  <Image source={{ uri: flower.image }} style={styles.selectedImage} />
                  <TouchableWithoutFeedback onPress={() => removeSelectedFlower(flower)}>
                    <MaterialCommunityIcons
                      style={styles.closeIcon}
                      name="close-circle"
                      size={20}
                      color="#000"
                    />
                  </TouchableWithoutFeedback>
                </Animated.View>
              )
            }

            return (
              <View key={flower._id} style={styles.circle}>
                <Image source={{ uri: flower.image }} style={styles.selectedImage} />
                <TouchableWithoutFeedback onPress={() => removeSelectedFlower(flower)}>
                  <MaterialCommunityIcons
                    style={styles.closeIcon}
                    name="close-circle"
                    size={20}
                    color="#000"
                  />
                </TouchableWithoutFeedback>
              </View>
            )
          })}
        </View>
        <View style={styles.bottomRight}>
          <TouchableOpacity onPress={() => navigate('ContactDetails')}>
            <FontAwesome name="arrow-right" size={40} color="#fff" />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
