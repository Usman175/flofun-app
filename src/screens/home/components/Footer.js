import React from 'react'
import { View, TouchableOpacity, Image, TouchableWithoutFeedback } from 'react-native'
import { FontAwesome, MaterialCommunityIcons } from '@expo/vector-icons'
import styles from './styles/Footer'

const Footer = ({ selectedFlowers, removeSelectedFlower, navigate }) => (
  selectedFlowers.length > 0 ?
  (
    <View style={styles.bottomContainer}>
      <View style={styles.bottomLeft}>
        {selectedFlowers.map((flower, id) =>
          <View key={id} style={styles.circle}>
            <Image source={{ uri: flower.image }} style={styles.selectedImage} />
            <TouchableWithoutFeedback onPress={() => removeSelectedFlower(flower)}>
              <MaterialCommunityIcons
                style={styles.closeIcon}
                name="close-circle"
                size={20}
                color="#000"
              />
            </TouchableWithoutFeedback>
          </View>
        )}
      </View>
      <View style={styles.bottomRight}>
        <TouchableOpacity onPress={() => navigate('ContactDetails')}>
          <FontAwesome
            name="arrow-right"
            size={40}
            color="#fff"
          />
        </TouchableOpacity>
      </View>
    </View>
  )
  : null
)

export default Footer
