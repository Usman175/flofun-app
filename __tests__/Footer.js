import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EStyleSheet from 'react-native-extended-stylesheet'
import Colors from '../constants/colors'
import Footer from '../src/commons/Footer'

beforeAll(() => {
  EStyleSheet.build(Colors)
})

describe('Footer', () => {
  test('renders correctly', () => {
    const footer = shallow(<Footer />)

    expect(toJson(footer)).toMatchSnapshot()
  })
})
