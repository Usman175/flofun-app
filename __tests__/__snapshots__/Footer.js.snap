// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#574E6D",
      "flex": 0.1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.8,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingLeft": 15,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0.2,
        "justifyContent": "space-around",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      focusedOpacity={0.7}
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        color="#fff"
        name="arrow-right"
        size={40}
      />
    </TouchableOpacity>
  </View>
</View>
`;
